### PR TITLE
fix!: unconfirmed pending cctx should prevent further outbound tx signing

### PR DESCRIPTION
**Pull request by brewmaster012**
# Description

1. Modify the AllPending RPC to return all pending txs
2. Bound the outtx tracker size
3. Allow add to outtx tracker only if a corresonding cctx is pending
4. A MsgAddToOutTxTracker can now remove an existing tracker if no pending cctx is found. 

related: #1255 

## upgrade compatibility

1. consensus breaking
2. no need for state migration when upgrade
3. (on athens3) need to cleanup old out tx trackers to not have performance issues

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: zetaclient/zetacore_observer.go [hunk 1]
```diff
@@ -178,6 +178,10 @@ func (co *CoreObserver) startSendScheduler() {
 								co.logger.ZetaChainWatcher.Error().Msgf("mismatch chainid: want %d, got %d", c.ChainId, params.ReceiverChainId)
 								continue
 							}
+							if params.OutboundTxTssNonce > cctxList[0].GetCurrentOutTxParam().OutboundTxTssNonce+120 {
```
Comment (lumtis) [inline zetaclient/zetacore_observer.go:181]: I think we should define a constant variable for the number `120`

File: proto/crosschain/tx.proto [hunk 2]
```diff
@@ -75,7 +75,9 @@ message MsgAddToOutTxTracker {
   int64 tx_index = 7;
 }
 
-message MsgAddToOutTxTrackerResponse {}
+message MsgAddToOutTxTrackerResponse {
+  bool is_removed = 1; // if the tx was removed from the tracker due to no pending cctx
```
Comment (lumtis) [inline proto/crosschain/tx.proto:79]: It seems it is not used in the message handler returns

Comment (brewmaster012) [reply proto/crosschain/tx.proto:79]: most cases defaults to false which is correct; 
there is one place it is set to true. 
Can add more explicit assignment but maybe not littering is better as the is_remove=true is really special case.

File: zetaclient/zetacore_observer.go (hunk 1 above)
Comment (brewmaster012) [reply zetaclient/zetacore_observer.go]: [1275932](https://github.com/zeta-chain/node/pull/1370/commits/127593241e54a40750fb9b14650bebe8683791a1)

File: proto/crosschain/tx.proto (hunk 2 above)
Comment (lumtis) [reply proto/crosschain/tx.proto:79]: I agree to not specify the value if it is false.
But it seems it is not used when it should be true: https://github.com/zeta-chain/node/pull/1370/files#diff-9ef5905d374779457a649b7c98c6b5ed348612bcb1a43a30d0923fecb27e86bfR64

**Review by ws4charlie — APPROVED** [review]
looks good

File: proto/crosschain/tx.proto (hunk 2 above)
Comment (brewmaster012) [reply proto/crosschain/tx.proto:79]: ah, yes
fixed here https://github.com/zeta-chain/node/pull/1370/commits/9edf3e9afc6772a3d747fc4ca7c1235286fb46c3

**Review by lumtis — APPROVED** [review]

**Pull request merged**